src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationService.cs

```diff
@@ -71,6 +71,38 @@ protected static bool IsNonIntersectingNamespace(ISymbol recommendationSymbol, S
                                               declarationSyntax.Span.IntersectsWith(candidateLocation.SourceSpan)));
         }
 
+        /// <summary>
+        /// If container is a tuple type, any of its tuple element which has a friendly name will cause
+        /// the suppression of the corresponding default name (ItemN).
+        /// In that case, Rest is also removed.
+        /// </summary>
+        protected static IEnumerable<ISymbol> SuppressDefaultTupleElements(INamespaceOrTypeSymbol container,
+            IEnumerable<ISymbol> symbols)
+        {
+            if (!container.IsType)
+            {
+                return symbols;
+            }
+
+            var type = (ITypeSymbol)container;
+            if (!type.IsTupleType)
+            {
+                return symbols;
+            }
+
+            var tuple = (INamedTypeSymbol)type;
```

Comment:
I.E.  something like:
```c#
   return symbols.Where(s => !s.IsImplicitlyDeclared && ((IFieldSymbol)s)?.TupleElementIndex >= 0) ;
```

For now it could be sufficient to do
```c#
   return symbols.Where(s => !s.IsImplicitlyDeclared && s.Name != "Rest") ;
```